airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/filters.py

```diff
@@ -90,5 +91,19 @@ def base64decode(value: str) -> str:
     return base64.b64decode(value.encode("utf-8")).decode()
 
 
-_filters_list = [hash, base64encode, base64decode]
+def string(value: Any) -> str:
```

Comment:
@girarda it looks like this change also means that the CDK will no longer complain when someone tries to pass in a non-stringified array value to a query parameter.

For example, when I try this on cloud, I get the error shown:
<img width="1508" alt="Screenshot 2024-03-22 at 5 12 00 PM" src="https://github.com/airbytehq/airbyte/assets/22731524/8339b5bd-ddf9-429e-9fdf-c1f84485c89e">

But when I try this on your branch, it lets me set it:
<img width="1529" alt="Screenshot 2024-03-22 at 5 12 14 PM" src="https://github.com/airbytehq/airbyte/assets/22731524/b843a4af-66db-4cae-8311-4de67a5e7e90">
and if I make it fail by giving it a bogus URL path (so that I can see the actual url-encoded request URL), I can see that it is actually doing the approach where it sets the same key multiple times:
<img width="1552" alt="Screenshot 2024-03-22 at 5 14 12 PM" src="https://github.com/airbytehq/airbyte/assets/22731524/f11f9f6a-9147-456b-984e-fd7ee3cc4b2a">

Maybe this is actually what we want? Because then for [this request](https://airbytehq-team.slack.com/archives/C03GD9SV36E/p1711040439412539?thread_ts=1710540234.262429&cid=C03GD9SV36E), the default behavior does what they want (splitting the values across multiple duplicate keys), but if someone doesn't want that behavior, they can do the `| string` macro that you added in this PR to treat the value as a raw string instead.

Reply:
support for passing the array as multiple query params was added yesterday because it was blocking a migration https://github.com/airbytehq/airbyte/pull/36357

Reply:
Ah I understand, thanks for clarifying